test/unit/worker_test.js

```diff
@@ -0,0 +1,228 @@
+      ];
+
+      return files;
+    };
+
+    const workerConfig = {
+      by: createTestGroups,
+      testConfig: './test/data/sandbox/codecept.customworker.js',
+    };
+
+    const workers = new Workers(-1, workerConfig);
+
+    for (const worker of workers.getWorkers()) {
+      worker.addConfig({
+        helpers: {
+          FileSystem: {},
+          Workers: {
+            require: './custom_worker_helper',
+          },
+        },
+      });
+    }
+
+    workers.run();
+
+    workers.on(event.all.result, (status) => {
+      expect(workers.getWorkers().length).equal(2);
+      expect(status).equal(true);
+      done();
+    });
+  });
+
+
+  it('should run worker with custom config', (done) => {
+    if (!semver.satisfies(process.version, '>=11.7.0')) this.skip('not for node version');
+
+    const workerConfig = {
+      by: 'test',
+      testConfig: './test/data/sandbox/codecept.customworker.js',
+    };
+    let passedCount = 0;
+    let failedCount = 0;
+
+    const workers = new Workers(2, workerConfig);
+
+    for (const worker of workers.getWorkers()) {
+      worker.addConfig({
+        helpers: {
+          FileSystem: {},
+          Workers: {
+            require: './custom_worker_helper',
+          },
+        },
+      });
+    }
+
+    workers.run();
+
+    workers.on(event.test.failed, () => {
+      failedCount += 1;
+    });
+    workers.on(event.test.passed, () => {
+      passedCount += 1;
+    });
+
+    workers.on(event.all.result, (status) => {
+      expect(status).equal(false);
+      expect(passedCount).equal(4);
+      expect(failedCount).equal(1);
+      done();
+    });
+  });
+
+
+  it('should able to add tests to each worker', (done) => {
+    if (!semver.satisfies(process.version, '>=11.7.0')) this.skip('not for node version');
+
+    const workerConfig = {
+      by: 'test',
+      testConfig: './test/data/sandbox/codecept.customworker.js',
+    };
+
+    const workers = new Workers(-1, workerConfig);
+
+    const workerOne = workers.spawn();
+    workerOne.addTestFiles([
+      path.join(codecept_dir, '/custom-worker/base_test.worker.js'),
+    ]);
+
+    const workerTwo = workers.spawn();
+    workerTwo.addTestFiles([
+      path.join(codecept_dir, '/custom-worker/custom_test.worker.js'),
+    ]);
+
+
+    for (const worker of workers.getWorkers()) {
+      worker.addConfig({
+        helpers: {
+          FileSystem: {},
+          Workers: {
+            require: './custom_worker_helper',
+          },
+        },
+      });
+    }
+
+    workers.run();
+
+    workers.on(event.all.result, (status) => {
+      expect(workers.getWorkers().length).equal(2);
+      expect(status).equal(true);
+      done();
+    });
+  });
+
+
+  it('should able to add tests to using createGroupsOfTests', (done) => {
+    if (!semver.satisfies(process.version, '>=11.7.0')) this.skip('not for node version');
+
+    const workerConfig = {
+      by: 'test',
```

Comment:
you don't need this by.test because in this example you split tests by suites...